src/services/store-service.js

```diff
@@ -31,19 +31,38 @@
                 });
             }
 
-            function save() {
+            function save(stock) {
                 store.set('stocks', favouriteStocks);
+                $rootScope.$broadcast('updateFavourites', stock);
             }
 
             function get() {
                 return order(favouriteStocks);
             }
 
+            // Move given item in an array to directly after the to-item
+            function reorder(fromItem, toItem) {
+                if (fromItem === toItem) {
+                    return;
+                }
+
+                var oldArray = order(favouriteStocks);
+                var fromIndex = oldArray.indexOf(fromItem);
+                var toIndex = oldArray.indexOf(toItem);
+                oldArray.splice(toIndex, 0, oldArray.splice(fromIndex, 1)[0]);
+
+                for (var i = 0, max = oldArray.length; i < max; i++) {
+                    favouriteStocks[oldArray[i]] = i;
+                }
+
+                save();
```

Comment:
I agree this looks odd (perhaps it's an anti-pattern). The `stock` input to the `save` function is the data that gets bound to the event emitted. This event is caught in the search controller and the bound stock data is used to find the correct search result and update its icon / remove it from the list.

In this case the stock's favourite status has not changed, only its order. Also, given at least two stocks' orders have changed it's not clear which stock should be passed in the event.

However, passing any stock (the one that was just moved feels most sensible) would probably be safe as the subscribers to the event wouldn't care.